pkg/nsx/services/subnetport/builder.go

```diff
@@ -242,7 +242,9 @@ func (service *SubnetPortService) buildExternalAddressBinding(sp *v1alpha1.Subne
 		return nil, nil
 	}
 	portExternalAddressBinding := &model.ExternalAddressBinding{}
-	if restoreMode && len(addressBinding.Status.IPAddress) > 0 {
+	if addressBinding.Spec.AllocatedExternalIPPath != "" {
+		portExternalAddressBinding.AllocatedExternalIpPath = &addressBinding.Spec.AllocatedExternalIPPath
```

Comment:
Should we need update https://github.com/vmware-tanzu/nsx-operator/blob/9e03ec05a85554c1c859a284e236411e48e12c63/pkg/nsx/services/subnetport/compare.go#L41? Maybe `s.ExternalAddressBinding = &model.ExternalAddressBinding{}` -> `s.ExternalAddressBinding = &model.ExternalAddressBinding{AllocatedExternalIpPath: sp.ExternalAddressBinding.AllocatedExternalIpPath}`?

Reply:
Updated this part to support update.